### ISSUE TITLE
Wrapped unions

### ISSUE THREAD

**Issue by nicosierra** [Vincit/objection.js#1057]
The following issue comes when trying to implement wrapped unions.

I need to produce the following query:
```sql
(select `user_id` from `someTableName` where `user_id` = ?  limit 1 ) 
UNION 
(select `user_id` from `anotherTableName` where `user_id` = ? limit 1 )
```

i.e. create unions that are wrapped from the beginning.

The `union` and `unionAll` methods, can wrap the subsequent queries, if the second parameter is set as `true`, but since the entrypoint to any queries in Objection is through the `model.query` method, it's hard to wrap anything before (please, I have tried with having a `raw` query in the beginning, but it looks hacky and it does not follow the same syntax of the subsequent queries.

It's a known issue on [knex](https://github.com/tgriesser/knex/issues/913), and it might be related to https://github.com/Vincit/objection.js/issues/842.

How to reproduce:

```
class AModel extends Model {
  static get tableName() {
    return 'someTableName';
  }
}

class AnotherModel extends Model {
  static get tableName() {
    return 'anotherTableName';
  }
}

AModel.query()
  .select('user_id')
  .where('user_id', '=', 1)
  .limit(1)
  .union(
    AnotherModel.query()
      .select('user_id')
      .where('user_id', '=', 1)
      .limit(1),
    true,
  )
  .debug()
  .then(r => r);
 ```

Generates the following sql statement:
```sql
select `user_id` from `someTableName` where `user_id` = ? union (select `user_id` from `anotherTableName` where `user_id` = ? limit ?) limit ?'
```
> \* so it puts the limit outside the union

I'd appreciate any help on how to implement this.
Objection-code-wise, I think that there could be an solution like a wrappedQuery perhaps? Not sure.
Cheers! 🙂

**Comment by nicosierra** [comment]
So far, this works. I wonder if we can stay with objection methods.

```
const queries = [
  AModel.query()
    .select('user_id')
    .where('user_id', '=', 1)
    .limit(1),
  AnotherModel.query()
    .select('user_id')
    .where('user_id', '=', 1)
    .limit(1),
];

knex.raw(queries.map(q => q.toSql()).join(' UNION ')).then(r => console.log(r));
```

**Comment by koskimas** [comment]
Do you really need to use union in this case? How about just two queries, and then combine the results? This should be fixed in knex, not objection.

**Comment by nicosierra** [comment]
@koskimas, thanks for your answer! sure that sounds way easier 😄 
I need to do that with 10 tables though, and at some point thought it would perform better with the union.

Whether to achieve performance by running single queries, all of them in a transaction, or within a UNION statement is a question I still have to go through.

Nonetheless, I was wondering how to achieve wrapped unions, since it is quite a plausible scenario, and maybe some people are also struggling to achieve that.

Thanks for your feedback!

**Comment by rosendi** [comment]
@nicosierra May be that could help you:

```
YourModel.knex()
  .unionAll(YourModel.knexQuery().limit(1), true)
  .unionAll(YourModel.knexQuery().where(...), true)
```

**Comment by koskimas** [comment]
@nicosierra With 10 tables you are probably better off with a union :smile: Unfortunately there is no way to currently do this and adding support would mean exposing a way to build a `knex` query from an objection query, which will currently have the same problems as `toString` and `toSQL` methods. This may be fixed in 2.0, but until then, you need to find some other way to achieve this.

**Comment by zsf3** [comment]
Hi @koskimas Just wanted to know if this was fixed in 2.0 or 3.0? If yes, then could you please share the link to documentation about how to overcome this scenario using unions? I also have a use case to fetch data from multiple (more than 5) tables and I was thinking to use unions and then came across this issue.
